packages/taquito/src/contract/rpc-estimate-provider.ts

```diff
@@ -127,16 +150,17 @@ export class RPCEstimateProvider extends OperationEmitter implements EstimationP
       throw new TezosOperationError(errors);
     }
 
-    while (
-      opResponse.contents.length !== (Array.isArray(params.operation) ? params.operation.length : 1)
-    ) {
-      opResponse.contents.shift();
-    }
+    let numberOfOps = Array.isArray(params.operation) ? params.operation.length : 1;
+    const containRevealOp = opResponse.contents[0].kind === 'reveal';
+    if (containRevealOp) {
+      numberOfOps--;
+    };
 
     return opResponse.contents.map(x => {
-      return this.createEstimateFromOperationContent(
+      return this.getEstimationPropetiesFromOperationContent(
         x,
-        opbytes.length / 2 / opResponse.contents.length,
+        // TODO: Calculate a specific opSize for each operation.
```

Comment:
We can create an issue and link it here to avoid losing track of this

Reply:
Makes sense, here is the issue: https://github.com/ecadlabs/taquito/issues/829